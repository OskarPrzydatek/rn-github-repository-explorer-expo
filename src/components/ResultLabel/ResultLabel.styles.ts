import { StyleSheet } from 'react-native';

import { colors } from '@/constants';

export const resultLabelStyles = StyleSheet.create({
  resultLabel: {
    alignItems: 'baseline',
    width: '80%',
  },
  divider: {
    paddingBottom: 8,
    borderBottomWidth: 2,
    borderBottomColor: colors.orange,
  },
  text: {
    fontWeight: 'bold',
  },
});
