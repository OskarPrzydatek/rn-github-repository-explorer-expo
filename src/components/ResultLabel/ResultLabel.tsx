import React from 'react';

import { Text, View } from 'react-native';

import { resultLabelStyles } from '@/components/ResultLabel/ResultLabel.styles';

interface IResultLabel {
  resultLabel: string;

  testID?: string;
}

export const ResultLabel = ({ resultLabel, testID }: IResultLabel) => {
  return (
    <View style={resultLabelStyles.resultLabel} testID={testID}>
      <View style={resultLabelStyles.divider}>
        <Text numberOfLines={1} style={resultLabelStyles.text}>
          {resultLabel}
        </Text>
      </View>
    </View>
  );
};
