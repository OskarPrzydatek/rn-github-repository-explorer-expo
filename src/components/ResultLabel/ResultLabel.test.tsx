import { render } from '@testing-library/react-native';

import { ResultLabel } from '@/components/ResultLabel/ResultLabel';

describe('ResultLabel', () => {
  it('should match snapshot', () => {
    const tree = render(<ResultLabel resultLabel={''} />);
    expect(tree).toMatchSnapshot();
  });
});
