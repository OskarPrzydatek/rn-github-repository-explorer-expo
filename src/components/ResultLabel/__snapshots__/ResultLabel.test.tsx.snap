// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultLabel should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "baseline",
      "width": "80%",
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#fd8c73",
        "borderBottomWidth": 2,
        "paddingBottom": 8,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        {
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
